docs/backend/testing/running-server-tests-inside-a-transaction.md

```diff
@@ -175,18 +175,14 @@ If you use `Suite.Run`:
 
 Sometimes, it's too much of refactor to pull out all the setup, or the setup takes a very long time.
 
-This is not a great situation, but if you still want to keep some of the transactional performance benefits, you can use an alternate `Run` function called `RunWithPreloadedData`. This is a **less ideal pattern** for the following reasons:
+In these cases, you can use `suite.PreloadData`. This can only be called once per test and will set up data that will then be used by all subtests. Once it is called, it will set a savepoint from which all subtests will start. When a subtests ends, the db will rollback to the savepoint. This way, each subtest can reuse the preloaded data, but subtests cannot modify the data used by another test. In most situations, it is preferrable to extract db setup into a setup function.
 
-* It doesn't actually use the newer txdb transactions
-* You cannot use this if the underlying code under test uses transactions.
-* If someone later changes the underlying code to use transactions, they will have to do the work to fully switch to `suite.Run`, the preferred pattern. This creates a deterrent to using transactions in the codebase, and we don't want to deter that.
-* It rolls back each subtest, but doesn't use a new connection. So any changes made in the main test will be available in the subtests.
-
-It can be preferred in one case:
+`PreloadData` can be preferred in one case:
 
 * Where there is a large amount of necessary setup, like populating the rate engine for payment calculations. In this case it can be valuable to have a preloaded setup.
 
-So **if you absolutely must** use `RunWithPreloadedData`, the process is easy. Just change the `suite.Run` calls to `suite.RunWithPreloadedData`.
+So **if you absolutely must** use `PreloadData`, the process is easy. Create a `suite.PreloadData` function and do all your test setup inside. [Here's an example](https://github.com/transcom/mymove/blob/c4f7df208f4b63fd85b0cd7645156a3516902027/pkg/rateengine/nonlinehaul_test.go#L188)
+
 
```

Comment:
Looks good! 

Can you add one more bullet to the "My tests started failing after making these changes"?

Basically we do tell them to watch out for `suite.DB()`, but I'd like to make it clear that any call to `suite.DB()`, **even if they are creating a stubbed object**, should be moved from the parent test. Because it's the call to `suite.DB()` that performs some setup that we should only do in certain locations.